src/main/java/com/ifmo/epampractice/dao/InvoiceDao.java

```diff
@@ -0,0 +1,130 @@
+package com.ifmo.epampractice.dao;
+
+import com.ifmo.epampractice.entity.InvoiceEntity;
+import com.ifmo.epampractice.enums.InvoiceStatus;
+
+import java.math.BigDecimal;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class InvoiceDao implements BasicDaoInterface<InvoiceEntity> {
+    static final String GET_ALL_QUERY = "SELECT * FROM INVOICE";
+    static final String GET_QUERY = "SELECT * FROM INVOICE WHERE ID = ?";
+    static final String SAVE_QUERY = "INSERT INTO INVOICE (ORDER_ID, ISSUE_DATE, " +
+            "PAYMENT_DATE, TOTAL_PRICE, STATUS) VALUES ( ?, ?, ?, ?, ?::e_status_invoice)";
+    static final String UPDATE_QUERY = "UPDATE INVOICE SET (ORDER_ID, ISSUE_DATE, " +
+            "PAYMENT_DATE, TOTAL_PRICE, STATUS) = (?, ?, ?, ?, ?::e_status_invoice) WHERE ID = ?";
+    static final String DELETE_QUERY = "DELETE FROM INVOICE WHERE ID = ?";
+
+    private InvoiceEntity parseRow(ResultSet resultSet) throws SQLException {
+        int id = resultSet.getInt("id");
+        int orderId = resultSet.getInt("order_id");
+        Date issueDate = resultSet.getDate("issue_date");
+        Date paymentDate = resultSet.getDate("payment_date");
+        BigDecimal totalPrice = resultSet.getBigDecimal("total_price");
+        String statusString = resultSet.getString("status").toUpperCase();
+        InvoiceStatus status = InvoiceStatus.valueOf(statusString);
+
+        return new InvoiceEntity(id, orderId, issueDate, paymentDate, totalPrice, status);
+    }
+
+    private void setInvoiceFieldsToStatement(InvoiceEntity invoice, PreparedStatement statement) throws SQLException {
+        statement.setInt(1, invoice.getOrderId());
+        statement.setDate(2, new java.sql.Date(invoice.getIssueDate().getTime()));
+        statement.setDate(3, new java.sql.Date(invoice.getPaymentDate().getTime()));
+        statement.setBigDecimal(4, invoice.getTotalPrice());
+        statement.setString(5, invoice.getStatus().name().toLowerCase());
+    }
+
+    @Override
+    public List<InvoiceEntity> getAll() {
+        List<InvoiceEntity> invoices = new ArrayList<>();
+        DBConnectorInterface dbConnector = DBConnectorPostgres.getInstance();
+        Connection connection = dbConnector.getConnection();
+        try {
+            Statement statement = connection.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE,
+                    ResultSet.CONCUR_READ_ONLY);
+            ResultSet resultSet = statement.executeQuery(GET_ALL_QUERY);
+            resultSet.first();
+            do {
+                InvoiceEntity currentInvoice = parseRow(resultSet);
+                invoices.add(currentInvoice);
+            }
+            while (resultSet.next());
```

Comment:
Check documentation on ResultSet.next().
TL;DR by default, result set points at nothing. Correct usage is `while (resultSet.next()) { ... }`, NOT do/while